File: Registration/Registration.Sample.VisualBasic/ExampleAddIn.vb
﻿Imports ExcelDna.Integration

Public Class ExampleAddIn
    Implements IExcelAddIn

    Public Sub AutoOpen() Implements IExcelAddIn.AutoOpen
        ExcelIntegration.RegisterUnhandledExceptionHandler(Function(ex) "!!! ERROR: " + ex.ToString())
    End Sub

    Public Sub AutoClose() Implements IExcelAddIn.AutoClose

    End Sub

End Class

File: Registration/Registration.Sample/ExampleAddIn.cs
using ExcelDna.Integration;

namespace Registration.Sample
{
    public class ExampleAddIn : IExcelAddIn
    {
        public void AutoOpen()
        {
            // First example if Instance -> Static conversion
            InstanceMemberRegistration.TestInstanceRegistration();
        }

        public void AutoClose()
        {
        }
    }
}
